File: src/main/scala/im/tox/toktok/app/profile/ProfileFragment.scala
package im.tox.toktok.app.profile

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.support.design.widget.CoordinatorLayout
import android.support.v4.app.Fragment
import android.support.v4.view.GravityCompat
import android.support.v7.app.AppCompatActivity
import android.view.ViewGroup.LayoutParams
import android.view.{ LayoutInflater, View, ViewGroup }
import im.tox.toktok.TypedResource._
import im.tox.toktok.app.simple_dialogs.{ SimpleShareDialogDesign, SimpleStatusDialogDesign, SimpleTextDialogDesign }
import im.tox.toktok.{ R, TR }

final class ProfileFragment extends Fragment {

  private implicit def activity: AppCompatActivity = getActivity.asInstanceOf[AppCompatActivity]

  override def onCreateView(inflater: LayoutInflater, container: ViewGroup, savedState: Bundle): CoordinatorLayout = {
    super.onCreate(savedState)
    val view = inflater.inflate(TR.layout.activity_profile, container, false)

    getActivity.getWindow.setStatusBarColor(Color.parseColor("#2b000000"))

    view.setLayoutParams(new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT))

    val toolbar = view.findView(TR.profile_toolbar)
    activity.setSupportActionBar(toolbar)
    activity.getSupportActionBar.setHomeAsUpIndicator(R.drawable.ic_navigation_menu)
    activity.getSupportActionBar.setDisplayHomeAsUpEnabled(true)

    val drawer = getActivity.findView(TR.home_layout)
    toolbar.setOnClickListener(new View.OnClickListener {
      override def onClick(v: View): Unit = {
        drawer.openDrawer(GravityCompat.START)
      }
    })

    val mShareIDButton = view.findView(TR.profile_share_id)
    mShareIDButton.setOnClickListener(new View.OnClickListener {
      override def onClick(v: View): Unit = {
        val dial = new SimpleShareDialogDesign(getActivity)
        dial.show()
      }
    })

    val mChangeNickname = view.findView(TR.profile_change_nickname)
    mChangeNickname.setOnClickListener(new View.OnClickListener {
      override def onClick(v: View): Unit = {
        val dial = new SimpleTextDialogDesign(
          getActivity,
          getResources.getString(R.string.profile_nickname),
          getResources.getColor(R.color.homeColorToolbar, null),
          R.drawable.ic_person_black_48dp,
          getResources.getString(R.string.sample_user_name),
          null
        )
        dial.show()
      }
    })

    val mChangeStatusMessage = view.findView(TR.profile_change_status_text)
    mChangeStatusMessage.setOnClickListener(new View.OnClickListener {
      override def onClick(v: View): Unit = {
        val dial = new SimpleTextDialogDesign(
          getActivity,
          getResources.getString(R.string.profile_status_message),
          getResources.getColor(R.color.homeColorToolbar, null),
          R.drawable.ic_person_black_48dp,
          getResources.getString(R.string.sample_status_message),
          null
        )
        dial.show()
      }
    })

    val mChangeStatus = view.findView(TR.profile_change_status)
    mChangeStatus.setOnClickListener(new View.OnClickListener {
      override def onClick(v: View): Unit = {
        val dial = new SimpleStatusDialogDesign(getActivity, 0)
        dial.show()
      }
    })

    val mChangeReject = view.findView(TR.profile_change_reject_call)
    mChangeReject.setOnClickListener(new View.OnClickListener {
      override def onClick(v: View): Unit = {
        startActivity(new Intent(activity, classOf[RejectedCallMessages]))
      }
    })

    view
  }

}

File: src/main/java/im/tox/toktok/app/profile/ProfileFragment.java
package im.tox.toktok.app.profile;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.CoordinatorLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewGroup.LayoutParams;
import android.widget.RelativeLayout;
import android.widget.TextView;

import im.tox.toktok.R;
import im.tox.toktok.app.simple_dialogs.SimpleShareDialogDesign;
import im.tox.toktok.app.simple_dialogs.SimpleStatusDialogDesign;
import im.tox.toktok.app.simple_dialogs.SimpleTextDialogDesign;

public final class ProfileFragment extends Fragment {

  @Override
  public CoordinatorLayout onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedState) {
    super.onCreate(savedState);
    CoordinatorLayout view = (CoordinatorLayout) inflater.inflate(R.layout.activity_profile, container, false);

    getActivity().getWindow().setStatusBarColor(Color.parseColor("#2b000000"));

    view.setLayoutParams(new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));

    final AppCompatActivity activity = (AppCompatActivity) getActivity();
    Toolbar toolbar = view.findViewById(R.id.profile_toolbar);
    activity.setSupportActionBar(toolbar);
    activity.getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_navigation_menu);
    activity.getSupportActionBar().setDisplayHomeAsUpEnabled(true);

    final DrawerLayout drawer = getActivity().findViewById(R.id.home_layout);
    toolbar.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        drawer.openDrawer(GravityCompat.START);
      }
    });

    RelativeLayout mShareIDButton = view.findViewById(R.id.profile_share_id);
    mShareIDButton.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        SimpleShareDialogDesign dial = new SimpleShareDialogDesign(getActivity());
        dial.show();
      }
    });

    TextView mChangeNickname = view.findViewById(R.id.profile_change_nickname);
    mChangeNickname.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        SimpleTextDialogDesign dial = new SimpleTextDialogDesign(
          getActivity(),
          getResources().getString(R.string.profile_nickname),
          getResources().getColor(R.color.homeColorToolbar, null),
          R.drawable.ic_person_black_48dp,
          getResources().getString(R.string.sample_user_name),
          null
        );
        dial.show();
      }
    });

    TextView mChangeStatusMessage = view.findViewById(R.id.profile_change_status_text);
    mChangeStatusMessage.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        SimpleTextDialogDesign dial = new SimpleTextDialogDesign(
          getActivity(),
          getResources().getString(R.string.profile_status_message),
          getResources().getColor(R.color.homeColorToolbar, null),
          R.drawable.ic_person_black_48dp,
          getResources().getString(R.string.sample_status_message),
          null
        );
        dial.show();
      }
    });

    TextView mChangeStatus = view.findViewById(R.id.profile_change_status);
    mChangeStatus.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        SimpleStatusDialogDesign dial = new SimpleStatusDialogDesign(getActivity(), 0);
        dial.show();
      }
    });

    TextView mChangeReject = view.findViewById(R.id.profile_change_reject_call);
    mChangeReject.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        startActivity(new Intent(getActivity(), RejectedCallMessages.class));
      }
    });

    return view;
  }

}
